test/users.test.js

```diff
@@ -0,0 +1,51 @@
+const request = require('supertest');
+
+const app = require('../app');
+// const { factoryByModel } = require('./factory/factory_by_models');
+const { createUser, buildUser } = require('./factory/users');
+
+let user = '';
+
+beforeEach(async () => {
+  user = await buildUser();
+});
+
+describe('Post Sign UP', () => {
+  test('Should sign up a new user', async done => {
+    // const user = await factoryByModel('User');
+    const response = await request(app)
+      .post('/users')
+      .send(user.dataValues);
+    expect(response.text).toContain('esteban.herrera');
+    done();
+  });
+
+  test('Should fail for invalid password', async done => {
+    const invalidPassword = { ...user.dataValues, password: 'contra12*' };
+    const response = await request(app)
+      .post('/users')
+      .send(invalidPassword)
+      .expect(422);
+    expect(response.text).toContain('password');
```

Comment:
It is better to use the `toBe` or `toEqual` assertions when you know exactly the phrase that is returned

Reply:
I use the ```toContain``` assertion because the response.text is something like this : 
``` "{\"message\":{\"password\":{\"value\":\"contra12*\",\"msg\":\"Password must be alphanumeric.\",\"param\":\"password\",\"location\":\"body\"}},\"internal_code\":\"schema_validation_error\"}" ```
Too long to use ```toBe``` or ```toEqual```

Reply:
Yeah, thats the thing. You should access the response through `response.body` not through `response.text`